airbyte-integrations/connectors/source-microsoft-teams/README.md

```diff
@@ -0,0 +1,100 @@
+to generate the necessary credentials. Then create a file `secrets/config.json` conforming to the `source_microsoft_teams/spec.json` file.
+See `sample_files/sample_config.json` for a sample config file.
+
+**If you are an Airbyte core member**, copy the credentials in RPass under the secret name `source-microsoft-teams-integration-test-config`
+and place them into `secrets/config.json`.
+
+
+### Locally running the connector
+```
+python main_dev.py spec
+python main_dev.py check --config secrets/config.json
+python main_dev.py discover --config secrets/config.json
+python main_dev.py read --config secrets/config.json --catalog sample_files/configured_catalog.json
+```
+
+### Unit Tests
+To run unit tests locally, from the connector directory run:
+```
+pytest unit_tests
+```
+
+### Locally running the connector docker image
+```
+# in airbyte root directory
+./gradlew :airbyte-integrations:connectors:source-microsoft-teams:airbyteDocker
+docker run --rm airbyte/source-microsoft-teams:dev spec
+docker run --rm -v $(pwd)/airbyte-integrations/connectors/source-microsoft-teams/secrets:/secrets airbyte/source-microsoft-teams:dev check --config /secrets/config.json
+docker run --rm -v $(pwd)/airbyte-integrations/connectors/source-microsoft-teams/secrets:/secrets airbyte/source-microsoft-teams:dev discover --config /secrets/config.json
+docker run --rm -v $(pwd)/airbyte-integrations/connectors/source-microsoft-teams/secrets:/secrets -v $(pwd)/airbyte-integrations/connectors/source-microsoft-teams/sample_files:/sample_files airbyte/source-microsoft-teams:dev read --config /secrets/config.json --catalog /sample_files/configured_catalog.json
+```
+
+### Integration Tests
+1. From the airbyte project root, run `./gradlew :airbyte-integrations:connectors:source-microsoft-teams:standardSourceTestPython` to run the standard integration test suite.
+1. To run additional integration tests, place your integration tests in the `integration_tests` directory and run them with `pytest integration_tests`.
+   Make sure to familiarize yourself with [pytest test discovery](https://docs.pytest.org/en/latest/goodpractices.html#test-discovery) to know how your test files and methods should be named.
+
+## Dependency Management
+All of your dependencies should go in `setup.py`, NOT `requirements.txt`. The requirements file is only used to connect internal Airbyte dependencies in the monorepo for local development.
+
+## App and Authentication
+
+The Microsoft Graph API uses OAuth for authentication. Follow these [instructions](https://docs.microsoft.com/en-us/graph/auth-v2-service?context=graph%2Fapi%2F1.0&view=graph-rest-1.0) for creating an app in the Azure portal. This process will produce the `client_id`, `client_secret`, and `tenant_id` needed for the tap configuration file.
+
+1. Login to [Azure Portal](https://portal.azure.com/#home)
```

Comment:
Could you add these instructions for generating the auth credentials in `docs/integrations/source/microsoft-teams.md` (a new file) and point to that URL from this section instead? The reason is: we should always centralized and DRY any documentation. Since we need these docs to be user facing anyways, we put the instructions for creating credentials in the `docs.airbyte.io` domain and point to it from this README.
 
Adding a file to `docs/integrations/source/microsoft-teams.md` will appear on https://docs.airbyte.io/integrations/sources/microsoft-teams when this PR is merged. 

See Braintree as an example: 
[connector readme](https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/source-braintree-singer)
[documentation](https://docs.airbyte.io/integrations/sources/braintree)

Furthermore if possible always point to an external link (in this case hosted in the microsoft teams website) because: 
1. if these instructions go out of date (e.g: microsoft changes their process for obtaining creds), we won't know that until someone reports an issue. Instead Microsoft will likely keep their own docs up to date. 
2. if we point to an external URL for docs, we will get an alert if the URL goes offline (we have automated checks for this in CI)